utils/loggers/wandb/wandb_utils.py

```diff
@@ -235,36 +277,65 @@ def setup_training(self, opt):
             if modeldir:
                 self.weights = Path(modeldir) / "last.pt"
                 config = self.wandb_run.config
-                opt.weights, opt.save_period, opt.batch_size, opt.bbox_interval, opt.epochs, opt.hyp, opt.imgsz = str(
-                    self.weights), config.save_period, config.batch_size, config.bbox_interval, config.epochs,\
-                    config.hyp, config.imgsz
+                (
+                    opt.weights,
+                    opt.save_period,
+                    opt.batch_size,
+                    opt.bbox_interval,
+                    opt.epochs,
+                    opt.hyp,
+                    opt.imgsz,
+                ) = (
+                    str(self.weights),
+                    config.save_period,
+                    config.batch_size,
+                    config.bbox_interval,
+                    config.epochs,
+                    config.hyp,
+                    config.imgsz,
+                )
         data_dict = self.data_dict
-        if self.val_artifact is None:  # If --upload_dataset is set, use the existing artifact, don't download
-            self.train_artifact_path, self.train_artifact = self.download_dataset_artifact(
-                data_dict.get('train'), opt.artifact_alias)
+        if (
+            self.val_artifact is None
+        ):  # If --upload_dataset is set, use the existing artifact, don't download
+            (
+                self.train_artifact_path,
+                self.train_artifact,
+            ) = self.download_dataset_artifact(
+                data_dict.get("train"), opt.artifact_alias
+            )
             self.val_artifact_path, self.val_artifact = self.download_dataset_artifact(
-                data_dict.get('val'), opt.artifact_alias)
+                data_dict.get("val"), opt.artifact_alias
+            )
 
         if self.train_artifact_path is not None:
-            train_path = Path(self.train_artifact_path) / 'data/images/'
-            data_dict['train'] = str(train_path)
+            train_path = Path(self.train_artifact_path) / "data/images/"
+            data_dict["train"] = str(train_path)
         if self.val_artifact_path is not None:
-            val_path = Path(self.val_artifact_path) / 'data/images/'
-            data_dict['val'] = str(val_path)
+            val_path = Path(self.val_artifact_path) / "data/images/"
+            data_dict["val"] = str(val_path)
 
         if self.val_artifact is not None:
-            self.result_artifact = wandb.Artifact("run_" + wandb.run.id + "_progress", "evaluation")
+            self.result_artifact = wandb.Artifact(
+                "run_" + wandb.run.id + "_progress", "evaluation"
+            )
             columns = ["epoch", "id", "ground truth", "prediction"]
-            columns.extend(self.data_dict['names'])
+            columns.extend([str(name) for name in self.data_dict["names"]])
```

Comment:
Changing the names to `str` from `int` when adding table columns overcomes the dtype issues we have for the fixes we make in the artifact logging